File: fproot/path.go
package main

import (
	"fmt"
	"strings"
)

func testPath(path string) (bool) {
	length := len(path)

	// Single slash test
	if length == 1 && path[0] == '/' {
		return true
	}

	// Local copy
	mypath := path

	// Strip trailing slash
	if path[(length-1)] == '/' {
		mypath = path[:(length-1)]
	}

	split := strings.Split(mypath, "/")[1:] // Uninterested in the first slash
	loc := 0

	for _, val := range split {
		if val == "." {
			continue
		} else if val == ".." {
			loc--
		} else {
			loc++
		}
	}

	if loc == 0 {
		return true
	}

	return false
}

func main () {
	singleSlash := testPath("/")
	folderTest := testPath("/bin/..")
	trickyTest := testPath("/./")
	expectedFalse := testPath("/bin/")

	fmt.Printf("/: %t, /bin/../ %t, /./: %t, /bin/: %t\n",
		singleSlash, folderTest, trickyTest, expectedFalse)
}

File: fproot/path.py
def test_path(path):
    """
    Takes a path and checks if it resolves to "/"

    /           -> True
    /./         -> True
    /<dir>/../  -> True
    /<anything> -> False
    """

    length = len(path)

    # Handle single slash
    if length == 1 and path == "/":
        return True

    # Strip trailing slash
    if path[-1] == "/":
        path = path[:-1]

    # Not portable to non-unix
    split = path.split("/")[1:]
    loc = []

    for thing in split:
        if thing == ".":
            # Same directory
            continue
        elif thing == "..":
            # One level up
            loc.pop()
        else:
            loc.append(thing)

    if len(loc) == 0:
        return True

    return False

